privaterelay/management/commands/update_phone_remaining_stats.py

```diff
@@ -3,58 +3,88 @@
 from django.conf import settings
 from django.core.management.base import BaseCommand
 
-from allauth.socialaccount.models import SocialAccount
+import logging
+
+from emails.models import Profile
+from privaterelay.management.utils import (
+    get_free_phone_social_accounts,
+    get_phone_subscriber_social_accounts,
+)
 
 if settings.PHONES_ENABLED:
     from phones.models import RelayNumber
 
-from privaterelay.views import update_fxa
+logger = logging.getLogger("events")
+
+
+def reset_phone_remaining_stats(user):
+    # re-set remaining_texts and remaining_seconds to the maximum value
+    try:
+        relay_number = RelayNumber.objects.get(user=user)
+    except RelayNumber.DoesNotExist:
+        # no RelayNumber set, do nothing
+        return
+    relay_number.remaining_texts = settings.MAX_TEXTS_PER_BILLING_CYCLE
+    relay_number.remaining_seconds = settings.MAX_MINUTES_PER_BILLING_CYCLE * 60
+    relay_number.save()
 
 
-def update_phone_remaining_stats():
-    if not settings.PHONES_ENABLED:
-        return []
-    accounts_with_phones = []
-    for sub_with_phone in settings.SUBSCRIPTIONS_WITH_PHONE:
-        social_accounts = SocialAccount.objects.filter(
-            extra_data__icontains=sub_with_phone
+def get_next_reset_date(profile: Profile) -> datetime:
+    # TODO: consider moving this as a property in Profile model
+    # assumes that profile being passed have already been checked to have
+    # phone subscription or a free phone user
+    if profile.date_phone_subscription_reset is None:
+        # there is a problem with the sync_phone_related_dates_on_profile
+        # or a new foxfooder whose date_phone_subscription_reset did not get set in
+        logger.error(
+            "phone_user_profile_dates_not_set",
+            extra={
+                "fxa_uid": profile.fxa.uid,
+                "date_subscribed_phone": profile.date_phone_subscription_end,
+                "date_phone_subscription_start": profile.date_phone_subscription_start,
+                "date_phone_subscription_reset": profile.date_phone_subscription_reset,
+                "date_phone_subscription_end": profile.date_phone_subscription_end,
+            },
```

Comment:
question (non-blocking): should we also put an error into sentry? what's the plan for addressing these errors?

Reply:
I'm hoping that this will tell us any holes in our logic so we can reproduce and fix them. Theoretically, the commands should cover for all scenarios but I'm always aware the reality is much bigger than the theory 🙃